File: Code Examples/Heavy/TimeoutExample.cs
//recursive solution of the knapsack problem (should timeout because of O(n^2))
class TimeoutExample
{
    public class Solution
    {
        public double FractionalKnapsack(Item[] items, int capacity)
        {            
            var sortedItems = items.OrderByDescending(i => i.Ratio).ToArray();

            return FractionalKnapsackRecursive(sortedItems, capacity, 0);
        }

        // Artificially exponential version (не оптимальный, просто ради нагрузки)
        private double FractionalKnapsackRecursive(Item[] items, int capacity, int index)
        {
            if (capacity == 0 || index == items.Length)
                return 0.0;

            double takeFull = 0;
            if (items[index].Weight <= capacity)
            {
                takeFull = items[index].Value + FractionalKnapsackRecursive(items, capacity - items[index].Weight, index + 1);
            }

            double takePartial = items[index].Ratio * Math.Min(capacity, items[index].Weight);

            double skip = FractionalKnapsackRecursive(items, capacity, index + 1);

            return Math.Max(Math.Max(takeFull, takePartial), skip);
        }

    }
}

File: Code Examples/Heavy/TimeoutExample.swift
class Solution {
    func fractionalKnapsack(_ items: [Item], _ capacity: Int) -> Double {
        let sortedItems = items.sorted { $0.ratio > $1.ratio }
        return fractionalKnapsackRecursive(sortedItems, capacity, 0)
    }

    private func fractionalKnapsackRecursive(_ items: [Item], _ capacity: Int, _ index: Int) -> Double {
        if capacity == 0 || index >= items.count {
            return 0.0
        }

        var takeFull = 0.0
        if items[index].weight <= capacity {
            takeFull = Double(items[index].value) + fractionalKnapsackRecursive(items, capacity - items[index].weight, index + 1)
        }

        let takePartial = items[index].ratio * Double(min(capacity, items[index].weight))
        let skip = fractionalKnapsackRecursive(items, capacity, index + 1)

        return max(takeFull, max(takePartial, skip))
    }
}